### PR TITLE
Poker/submit report not hardcoded

### PR DESCRIPTION
**Pull request by darbrius**

### CODE REVIEW

File: server/controllers/DataAccessLayer.js [hunk 1]
```diff
@@ -226,11 +227,12 @@ exports.ReadReportsFile = function() {
...
             // Filter and grab the username and message
-            let username = chatMessages[i].split(/,(.+)/)[0]
+            let userID = chatMessages[i].split(/,(.+)/)[0];
+            let user = cachedUsers.find(x => x.id === userID);
             let message = chatMessages[i].split(/,(.+)/)[1];
 
-            // Create a ChatMessage object
-            messages.push(new ChatMessage(username, message));
+            // Create a ChatMessage object with an empty tableID
+            messages.push(new ChatMessage("", user, message));
```
Comment (devongillis) [inline server/controllers/DataAccessLayer.js:235]: just to be clear, reports doesn't care about the table id?

Comment (darbrius) [reply server/controllers/DataAccessLayer.js:235]: Nope, the tableID is only relevant when storing chatMessages in the cache, and submitting reports. Once a report is submitted, the tableID isn't needed

Comment (devongillis) [reply server/controllers/DataAccessLayer.js:235]: gotcha, everything else looks good to me

**Review by devongillis — APPROVED** [review]

File: poker-pals/src/components/pokerComponents/ReportPocket.vue [hunk 2]
```diff
@@ -241,7 +241,8 @@ export default {
             }
             else {
                 // Send a filled report to the parent for submission to server
-                this.$parent.submitReport(this.selected, this.message, this.report_OffenderName, this.report_OffenderMessageId);
+                let submittingUser = this.$parent.userID;
```
Comment (rzayevan) [inline poker-pals/src/components/pokerComponents/ReportPocket.vue:244]: Where are you getting this id from? Poker component isn't passing userID though

File: server/controllers/DataAccessLayer.js [hunk 3]
```diff
@@ -330,7 +330,7 @@ GetChatLogString = function(chatMessages) {
     // For each ChatMessage in the given Report's chatLog
     for (let i = 0; i < chatMessages.length; i++) {
         // Create a unique string for the message
-        let chatMessageString = "{" + chatMessages[i].username + "," + chatMessages[i].message + "}";
+        let chatMessageString = "{" + chatMessages[i].userID + "," + chatMessages[i].message + "}";
```
Comment (rzayevan) [inline server/controllers/DataAccessLayer.js:333]: Where is this ID being retrieved from? I've made some changes around chat ids so might be good to take a look at my PR

File: poker-pals/src/components/pokerComponents/ReportPocket.vue (hunk 2 above)
Comment (darbrius) [reply poker-pals/src/components/pokerComponents/ReportPocket.vue:244]: this.$parent.userID accesses the Poker component and grabs the id of the current user.

File: server/controllers/DataAccessLayer.js (hunk 3 above)
Comment (darbrius) [reply server/controllers/DataAccessLayer.js:333]: The ID comes from the ChatMessage object, which gets the relevant id from the chat logs

File: poker-pals/src/components/pokerComponents/ReportPocket.vue (hunk 2 above)
Comment (rzayevan) [reply poker-pals/src/components/pokerComponents/ReportPocket.vue:244]: Yeaaaah, this has changed in my PR

File: server/controllers/DataAccessLayer.js (hunk 3 above)
Comment (rzayevan) [reply server/controllers/DataAccessLayer.js:333]: Nevermind, we're all good then

File: poker-pals/src/components/pokerComponents/ReportPocket.vue (hunk 2 above)
Comment (rzayevan) [reply poker-pals/src/components/pokerComponents/ReportPocket.vue:244]: You might need to do `this.$parent.userData.id` depending on who merges first

**Pull request merged**